components/SignInForm.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+
+const renderField = ({ input, label, type, meta: { touched, error } }) =>
+  <div>
+    <input {...input} placeholder={label} type={type} />
+    {touched &&
+      error &&
+      <div>
+        {error}
+      </div>}
+  </div>;
+
+const renderErrors = errors =>
+  <div className="alert alert-danger" role="alert">
+    {errors &&
+      errors.map(err =>
+        <span key={shortid.generate()}>
```

Comment:
You don't need to use shortid. You can get index from arguments. for example; (err, idx) => .....

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map